homeassistant/components/enphase_envoy/__init__.py

```diff
@@ -64,4 +64,11 @@ async def async_remove_config_entry_device(
         for inverter in envoy_data.inverters:
             if str(inverter) in dev_ids:
                 return False
+    if envoy_data and envoy_data.encharge_inventory:
+        for encharge in envoy_data.encharge_inventory:
+            if str(encharge) in dev_ids:
+                return False
+    if envoy_data and envoy_data.enpower:
+        if str(envoy_data.enpower.serial_number) in dev_ids:
+            return False
```

Comment:
Can we maybe look into automatic deletion at some point?

Reply:
This was added (without the encharge and enpower that are now added) in #74012, a year before the overhaul of the integration. So I assume this one was needed then, as confirmation a device could be manually removed or not. I think it's only used to prevent the user to remove a device manually if there's still an envoy, inverters, encharge and enpower present. An actual delete will occur when removing the integration entry.

So something automatically can be done in a future PR but I think till now we only allow disable.

Reply:
Yes this code allows to delete manually. But if we can know from the API for sure that a device is gone, we can do that automatically

Reply:
Ok, I'll add it to my to-do list for an upcoming PR. Adding the encharge and enpower test is to prevent an accidental manual delete for now.